gandalf.js

```diff
@@ -8,115 +8,112 @@
  *
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?